File: src/main/java/org/example/chessplatformbe/domain/Stream.java
package org.example.chessplatformbe.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Stream {

    private Integer id;

    private String name;

    private LocalDateTime creationDateTime;

    private String streamUrl;

    private Integer streamerId;
}

File: src/domain/stream.rs
use chrono::NaiveDate;

#[derive(Debug)]
pub struct Stream {
    id: i32,
    name: String,
    creation_date_time: NaiveDate,
    stream_url: String,
    streamer_id: i32,
}

impl Stream {
    pub fn new(
        id: i32,
        name: String,
        creation_date_time: NaiveDate,
        stream_url: String,
        streamer_id: i32,
    ) -> Self {
        Self {
            id,
            name,
            creation_date_time,
            stream_url,
            streamer_id,
        }
    }
}
